src/System.ServiceModel.Tests.Common/src/BridgeTestFixture.cs

```diff
@@ -0,0 +1,148 @@
+            {
+                resourceAddress = MakeResourcePutRequest(resourceName);
+                _Resources.Add(resourceName, resourceAddress);
+            }
+
+            return resourceAddress;
+        }
+
+        public static string GetBaseAddress(string resourceName, string name)
+        {
+            string resourceAddress = null;
+            if (_BridgeStatus == BridgeState.Started)
+            {
+                if (_BaseAddresses == null)
+                {
+                    _BaseAddresses = new Dictionary<string, string>();
+                    string response = MakeResourceGetRequest(resourceName);
+
+                    var reader = new JsonTextReader(new StringReader(response));
+                    while (reader.Read())
+                    {
+                        if (reader.TokenType == JsonToken.String)
+                        {
+                            _BaseAddresses.Add(reader.Path, reader.Value as string);
+                        }
+                    }
+                }
+
+                _BaseAddresses.TryGetValue(name, out resourceAddress);
+            }
+
+            return resourceAddress;
+        }
+
+        private static void MakeConfigRequest()
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.BaseAddress = new Uri(BridgeBaseAddress);
+                var content = new StringContent(
+                    @"{ resourcesDirectory : ""WcfService"" }",
+                    Encoding.UTF8,
+                    "application/json");
+                try
+                {
+                    var response = httpClient.PostAsync("/config/", content).Result;
+                    if (!response.IsSuccessStatusCode)
+                        throw new Exception("Unexpected status code: " + response.StatusCode);
+                    _BridgeStatus = BridgeState.Started;
+                }
+                catch (Exception exc)
+                {
+                    _BridgeStatus = BridgeState.Faulted;
+                    throw new Exception("Bridge is not running", exc);
+                }
+            }
+        }
+
+        private static string MakeResourcePutRequest(string resourceName)
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.BaseAddress = new Uri(BridgeBaseAddress);
+                var content = new StringContent(
+                        string.Format(@"{{ name : ""{0}"" }}", resourceName),
+                        Encoding.UTF8,
+                        "application/json");
+                try
+                {
+                    var response = httpClient.PutAsync("/resource/", content).Result;
+                    if (!response.IsSuccessStatusCode)
+                        throw new Exception("Unexpected status code: " + response.StatusCode);
```

Comment:
I think I see that the Bridge will return a 500 with a more informative body that described the problem. It might be useful to deserialize any response body and put it in the exception message.
Not required for this PR.